### DISCUSSION TITLE
Is it possible to make a custom connector to get token from Azure AD Application?

### DISCUSSION THREAD

**Discussion by sekharmalla** [microsoft/PowerPlatformConnectors#2032] [Q&A]
Hello, fellow Independent Publishers,

Hope everyone is doing fine,

Currently, I am working on a solution and I just wanted to know that is it possible for I can make a custom connector to get access_token from the Azure Access directory. I want to use the access token from the action response for another action. Example scenario:- 
![image](https://user-images.githubusercontent.com/43902003/194211627-5badcb17-987f-4b2b-96dc-5338b669df90.png)

Thanks in advance for taking your time,
Regards,
Chandra

**Comment by troystaylor** [comment]
It is certainly possible. You will create your connector like in the docs (https://learn.microsoft.com/en-us/connectors/custom-connectors/create-custom-connector-aad-protected-azure-functions#create-a-custom-connector) and then before submitting it, replace the OAuth credentials you have used with placeholder values. Can you tell me which Azure service you are trying to connect to? Please be aware that 1st-party Azure service connectors do take significantly longer to certify.

**Comment by sekharmalla** [comment]
Hello @troystaylor, Thanks for your reply, actually this is not for certification, I am making a Power platform solution and I will be taking Azure Ad application ID, Tenant ID, and Secret Key to get an access token then I will use that key for next action. I believe you are talking about authenticating with the Azure AD application(OAuth), but what I am looking for is an action that can take application objects and respond with access token.

**Comment by troystaylor** [comment]
Oh, sorry for my misunderstanding. I believe you can do this by using connection parameters and a policy, as roughly explained here: https://learn.microsoft.com/en-us/connectors/custom-connectors/policy-templates/setqueryparameter/setqueryparameter

**Comment by sekharmalla** [comment]
Okay, I will try to use connection parameters, but I am just a little bit confused, I will pass App ID, Tenant ID, and secret key from power apps to power automate then they will be passed onto the custom connector action. as you know we have to pass application objects in the request body, hence I set authentication to none. Do I need to set authentication as OAuth? If yes How can I pass Application objects dynamically everytime?

**Comment by troystaylor** [comment]
Can you detail why the key is being passed from your app instead of as an environmental or Flow variable? Do you have multiple apps trying to use the same Flow?
You can use connection parameters without authentication. You have to manually add them after downloading the connector and then update the connector back to your environment.

**Comment by sekharmalla** [comment]
Sure, I will tell you. We have the application and when a user Importing the solution into the new environment/Tenant there is a splash screen That will ask the user to provide Application Objects and then we will pass them to a flow which then will use them in an action and produce access token after a token is being received We use that token in for next actions. So this only happens one time, and will not be repeated once done. Unless the Admin user wants to add another application to authenticate. So If the solution is being Imported into a new Env/tenant only this is required. Hope that gives some idea of what I am trying to do. and the key is not being asked in the environment variable, eventually, we will be storing it in the env variable only but through the flow. to be precise When passing the application detailed on the splash screen and flow returns an error the splash screen should not go to the next screen but throw a notification to provide the correct details.

Appreciate your time and patience Thanks @troystaylor :)

**Reply by tech7857** [reply]
@sekharmalla Hi Sekhar were you able to achieve this? I have the same use case and looking for the solutions.

In my case only the client id and secret will change.

thanks
